kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -124,24 +125,75 @@ function _showRecentChannels(store, classId) {
 
 
 function _showRecentReports(store, classId, channelId) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
-  store.dispatch('SET_PAGE_NAME', Constants.PageNames.RECENT);
-
   // should be cached if navigated to this point
   const channelPromise = ChannelResource.getModel(channelId).fetch();
+  function __reportPropsGen(report) {
+    function ___descriptor(kind) {
+      switch (kind) {
+        case ContentNodeKinds.AUDIO:
+          return 'Listened';
+        case ContentNodeKinds.DOCUMENT:
+          return 'Opened';
+        case ContentNodeKinds.VIDEO:
+          return 'Watched';
+        case ContentNodeKinds.EXERCISE:
+          return 'Mastered';
+        case ContentNodeKinds.HTML5:
+          return 'Mastered';
+        default:
+          return 'Mastered';
+      }
+    }
+    const progress = report.progress[0];
+    const completed = `${progress.log_count_complete}/${progress.log_count_total}`;
+    const descriptor = ___descriptor(report.kind);
 
+    return {
+      completed,
+      descriptor,
+    };
+  }
   channelPromise.then(
     channelData => {
+      const sevenDaysAgo = new Date();
+      // this is being set by default in the backend
+      // backend date data might be unreliable, though
+      sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);
+
       const reportPayload = {
         channel_id: channelId,
         content_node_id: channelData.root_pk,
         collection_kind: ReportConstants.UserScopes.CLASSROOM,
         collection_id: classId,
+        last_active_time: sevenDaysAgo,
       };
       const recentReportsPromise = RecentReportResource.getCollection(reportPayload).fetch();
 
       recentReportsPromise.then(
         reports => {
+          // add progressProps to each report, for string generation
+          reports.forEach(
+            report => {
+              report.reportProps = __reportPropsGen(report);
+            }
+          );
+
+          // sort the reports in order of most recent
+          // might want to put this together with the above loop
+          reports.sort(
+            (report1, report2) => {
```

Comment:
Ah cool, I'll give it a shot. I did it this way because I remembered a chrome compatibility issue from another sort I did a while back. I looked it up, but it turns out I was using `1` and `0` via bool values:
https://github.com/learningequality/kolibri/commit/8e85303c26334f25fbe2e9f8fb994bdce3bbe857
Thanks!